wren-ui/src/apollo/server/resolvers/projectResolver.ts

```diff
@@ -38,6 +42,35 @@ export class ProjectResolver {
     this.startSampleDataset = this.startSampleDataset.bind(this);
   }
 
+  public async getSettings(_root: any, _arg: any, ctx: IContext) {
+    const project = await ctx.projectService.getCurrentProject();
+    const dataSourceType = project.type;
+
+    return {
+      dataSource: {
+        type: dataSourceType,
+        properties: this.getDataSourceProperties(project),
+        sampleDataset: project.sampleDataset,
+      },
+    };
+  }
+
+  public async resetCurrentProject(_root: any, _arg: any, ctx: IContext) {
+    try {
+      const project = await ctx.projectService.getCurrentProject();
+      await ctx.deployService.deleteAllByProjectId(project.id);
+      await ctx.askingService.deleteAllByProjectId(project.id);
+      await ctx.modelService.deleteAllViewsByProjectId(project.id);
+      await ctx.modelService.deleteAllModelsByProjectId(project.id);
+
+      await ctx.projectService.deleteProject(project.id);
+    } catch {
+      // do nothing
+    }
```

Comment:
a little bit weird that we ignore errors here

Reply:
It just needs to prevent the no project situation when doing reset,
I changed it back to this.
```ts
let project: Project;
try {
  project = await ctx.projectService.getCurrentProject();
} catch {
  // no project found
  return true;
}
```